chore(build): integrate ecommerce commons library (#5)
pagopa/pagopa-ecommerce-helpdesk-service: README.md

| Variable name                  | Description                                                       | type              | default |
|--------------------------------|-------------------------------------------------------------------|-------------------|---------|
| MONGO_HOST                     | Host where MongoDB instance used to persist wallet data           | hostname (string) |         |
| MONGO_PORT                     | Port where MongoDB is bound to in MongoDB host                    | number            |         |
| MONGO_USERNAME                 | MongoDB username used to connect to the database                  | string            |         |
| MONGO_PASSWORD                 | MongoDB password used to connect to the database                  | string            |         |
| MONGO_SSL_ENABLED              | Whether SSL is enabled while connecting to MongoDB                | string            |         |

### Run docker container

```shell
$ docker compose up --build
```

---

## Develop Locally 💻

### Prerequisites

- git
- gradle
- jdk-17
- ecommerce-commons library installed into maven local folder

### Run the project

```shell
$ ./gradlew bootRun
```

> ### Install ecommerce commons library locally

Reviewer: ```suggestion
### Install eCommerce commons library locally
```

Reviewer: Or
```suggestion
### Install `ecommerce-commons` library locally
```

> Into gradle build file have been added a task that take cares for you of proper ecommerce-commons library build,
> performing ecommerce-commons repository clone, checkout to the referred version and build (using maven).

Reviewer: ```suggestion
There is a task into the Gradle build file that take cares for you of properly fetching and building `ecommerce-commons`. It does so by performing a repository clone, checking out to the version set into the build file and building the library with Maven.
```

> If you want to re-build ecommerce commons library you can run build command with the `-PbuildCommons` argument

Reviewer: ```suggestion
If you want to re-build `ecommerce-commons` library you can run the build command with a `-PbuildCommons`.
```

> This two properties maps ecommerce commons version and ref branch:

Reviewer: ```suggestion
This two properties maps `ecommerce-commons` version and git ref:
```

````
val ecommerceCommonsVersion = "x.y.z" -> valued with ecommerce commons wanted pom version
val ecommerceCommonsRef = ecommerceCommonsVersion -> the branch/tag to be checkout.
````

> `ecommerceCommonsRef` is, by default, valued with the ecommerce commons library version, causing tagged version to be
> checkout. 

Reviewer: ```suggestion
`ecommerceCommonsRef` has by default the same value as `ecommerceCommonsVersion`, so that version tagged with `"x.y.z"` will be checked out and installed locally.  
```

This value was left as a separate property because, during developing phases can be changed to a feature branch
making the local build use a ref branch other than a tag for developing purpose.

```Shell
$ ./gradlew build -PbuildCommons
```

> Running the above command the version above task will run before project compilation building ecommerce commons locally inside maven local repository

Reviewer: ```suggestion
Running the above command the version above task will run before project compilation building eCommerce commons locally inside maven local repository
```

Reviewer: ```suggestion
Running the above command the version above task will run before project compilation building `ecommerce-commons` locally inside Maven local repository
```

> # PagoPA Help desk service

Reviewer: ```suggestion
# pagoPA Help desk service
```

[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=pagopa_pagopa-ecommerce-helpdesk-service&metric=alert_status)](https://sonarcloud.io/dashboard?id=pagopa_pagopa-ecommerce-helpdesk-service)

This microservice is responsible for ...

> - [PagoPA Help Desk Service](#pagopa-helpdesk-service)

Reviewer: ```suggestion
- [pagoPA Help Desk Service](#pagopa-helpdesk-service)
```
